src/main/java/pl/sda/weather/service/impl/LocationServiceImpl.java

```diff
@@ -1,30 +1,131 @@
 package pl.sda.weather.service.impl;
 
+import pl.sda.weather.Gui;
 import pl.sda.weather.repository.ILocationRepository;
 import pl.sda.weather.repository.impl.LocationRepositoryImpl;
 import pl.sda.weather.model.LocationModel;
 import pl.sda.weather.service.ILocationService;
 
+import java.util.ArrayList;
 import java.util.List;
 
 
 public class LocationServiceImpl implements ILocationService {
 
-    ILocationRepository locationDAO = new LocationRepositoryImpl();
+    ILocationRepository locationRepository = new LocationRepositoryImpl();
+
+    Gui gui = new Gui();
+
+    @Override
+    public void cleanDBWithLocalModel() {
+        this.locationRepository.cleanFile();
+    }
 
-@Override
-public void cleanFile(){
-    this.locationDAO.cleanFile();
-}
 
     @Override
-    public void addLocationModelJsonToDB(LocationModel locationModel){
-        this.locationDAO.addLocationModelJsonToDB(locationModel);
+    public void addLocationModelToDB(LocationModel locationModel) {
+        this.locationRepository.addLocationModelToDB(locationModel);
     }
+
     @Override
-    public List<LocationModel> getLocationModelFromFileJson() {
-        return this.locationDAO.getLocationModelFromFileJson();
+    public List<LocationModel> getLocationModelFromBD() {
+        return this.locationRepository.getLocationModelDataFromDB();
+
+
+    }
+
+    @Override
+    public List<LocationModel> getLocationModelFromDbAfterIdOrName(String patternToSearch) {
```

Comment:
As before, in my opinion this method shouldn't return List<LocationModel>, but just LocationModel